src/utils/sets.ts

```diff
@@ -32,3 +34,16 @@ export function setHasDiff<T>(a: Set<T>, b: Set<T>): boolean {
         return true; // different lengths means they are naturally diverged
     }
 }
+
+type Diff<T> = { added: T[], removed: T[] };
```

Comment:
can we raise this type so `arrayDiff` can use it too?

or, better yet, move it to the arrays.ts and export it from there. Then return it and have the `setDiff` for sets return `keyDiff` as-is, using the imported type.